### PR TITLE
feat: included the location in csv file

### LINKED ISSUE #1171: adding the location of GPS in the csv
**Actual Behaviour**

The GPS is not included in the data set.

**Expected Behaviour**

The GPS location should be fetched continuously when the logging starts and improved everytime a more precise location fetched and when the logging stops it should be included in the logged data along with the timestamp.

**Steps to reproduce it**

- Fetch location from GPS logger class [PR - #1157]
- Update the location when more precise is fetched 
- Include the location in the CSV file

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Avjeet**
Fixes #1171 
**Changes**: 
- Modified the GPS logger completely 
- Added dynamic check for permmision
- Added dynamic check for GPS 
- inlcuded the current location of user fetched by gps at the end of the csv file.

**Screenshot/s for the changes**: 
image

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[loc_data_csv.zip](https://github.com/fossasia/pslab-android/files/2168233/loc_data_csv.zip)

### CODE REVIEW

**Comment by CloudyPadmal** [comment]
@Avjeet nice! You may update this PR.

**Comment by Avjeet** [comment]
@CloudyPadmal conflicts resolved

File: app/src/main/java/org/fossasia/pslab/activity/LuxMeterActivity.java [hunk 1]
```diff
@@ -197,4 +211,44 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
+            if (gpsLogger.isGPSEnabled()) {
+                saveData = true;
+                Toast.makeText(getApplicationContext(), getString(R.string.data_recording_start),
+                        Toast.LENGTH_SHORT).show();
+            } else {
+                saveData = false;
+                Toast.makeText(getApplicationContext(), getString(R.string.gps_not_enabled),
+                        Toast.LENGTH_SHORT).show();
+            }
+
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/activity/LuxMeterActivity.java]: This line can be removed

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -908,5 +908,8 @@
     <string name="lux_meter_desc">\u2022 The above pin configuration has to be same except for the pin
     GND. GND is meant for Ground and any of the PSLab device GND pins can be used since they are common.\n\n
     \u2022 Select sensor by going to the Configure tab from the bottom navigation bar and choose BHT-1750 in the drop down menu under Select Sensor.\n</string>
+    <string name="allow_gps">Allow GPS</string>
+    <string name="allow_gps_info">Please turn on GPS to include the current location in the logged file</string>
+    <string name="gps_not_enabled">GPS not enabled</string>
```
Comment (CloudyPadmal) [inline app/src/main/res/values/strings.xml]: Place these strings along with other strings related to data logging

**Comment by Avjeet** [comment]
@CloudyPadmal changes done

**Pull request merged**